src/Raku/ast/statementprefixes.rakumod

```diff
@@ -617,8 +617,56 @@ class RakuAST::StatementPrefix::Phaser::Block
 # The FIRST phaser.
 class RakuAST::StatementPrefix::Phaser::First
   is RakuAST::StatementPrefix::Phaser::Block
+  is RakuAST::BeginTime
 {
     method type() { "FIRST" }
+
+    method is-begin-performed-before-children { True }
+    method is-begin-performed-after-children  { False }
+
+    method PERFORM-BEGIN(Resolver $resolver, RakuAST::IMPL::QASTContext $context) {
+        my $blorst := nqp::getattr(self, RakuAST::StatementPrefix, '$!blorst');
+
+        my $True := RakuAST::Term::Name.new(RakuAST::Name.from-identifier('True'));
+
+        my $attach-block := $resolver.find-attach-target('block');
```

Comment:
This is wrong. The only place where it's legal to ask for an attach target is the attach method. You don't know when PERFORM-BEGIN is called and what state the resolver will be in at that time. It could be that parsing some descendant is what triggers BEGIN handling and then your attach target may even be one of your children.